api/http/handler/endpointedge/async.go

```diff
@@ -0,0 +1,312 @@
+package endpointedge
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	httperror "github.com/portainer/libhttp/error"
+	"github.com/portainer/libhttp/request"
+	"github.com/portainer/libhttp/response"
+	portainer "github.com/portainer/portainer/api"
+	"github.com/portainer/portainer/api/internal/endpointutils"
+	"github.com/sirupsen/logrus"
+)
+
+// https://datatracker.ietf.org/doc/html/rfc6902
+// Doing this manually, because at this point, i don't want to marshal to json, make a diff - for now, i'm just using 'add'
+type JSONPatch struct {
+	Operation string      `json:"op"`
+	Path      string      `json:"path"`
+	Value     interface{} `json:"value"`
+}
+
+// TODO: copied from edgestack_status_update
+type updateStatusPayload struct {
+	Error      string
+	Status     *portainer.EdgeStackStatusType
+	EndpointID *portainer.EndpointID
+}
+type edgeJobResponse struct {
+	// EdgeJob Identifier
+	ID portainer.EdgeJobID `json:"Id" example:"2"`
+	// Whether to collect logs
+	CollectLogs bool `json:"CollectLogs" example:"true"`
+	// A cron expression to schedule this job
+	CronExpression string `json:"CronExpression" example:"* * * * *"`
+	// Script to run
+	Script string `json:"Script" example:"echo hello"`
+	// Version of this EdgeJob
+	Version int `json:"Version" example:"2"`
+}
+
+// An empty request ~~ just a ping.
+type Snapshot struct {
+	StackStatus map[portainer.EdgeStackID]updateStatusPayload
+}
+type AsyncRequest struct {
+	CommandId   string      `json: optional`
+	Snapshot    interface{} `json: optional` // todo
+	StackStatus map[portainer.EdgeStackID]updateStatusPayload
+}
+
+func (payload *AsyncRequest) Validate(r *http.Request) error {
+	// TODO:
+
+	return nil
+}
+
+type AsyncResponse struct {
+	CommandInterval  time.Duration `json: optional`
+	PingInterval     time.Duration `json: optional`
+	SnapshotInterval time.Duration `json: optional`
+
+	ServerCommandId      string      // should be easy to detect if its larger / smaller:  this is the response that tells the agent there are new commands waiting for it
```

Comment:
if this needs to be easily sorted/comparable, why not use an `int`instead of `string`?